addons/suggest_locations/models/suggest_by_product.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+from odoo import models, fields, _
+from itertools import groupby
+from .suggest_locations_policy import SuggestLocationPolicy, get_selection
+
+
+class ByProduct(SuggestLocationPolicy):
+    """Find locations based on other placed that product is stored"""
+
+    @classmethod
+    def name(cls):
+        return "by_product"
+
+    def get_values_from_mls(self, mls):
+        return {
+            "product": mls.product_id.ensure_one(),
+            "parent_location": mls.u_picking_type_id.default_location_dest_id.ensure_one(),
+        }
+
+    def _get_product_from_dict(self, values):
+        """Get the product from the values"""
+        product_id = values.get("product_id")
```

Comment:
If this is coming from the controller, is it an int ? or string? do we have to cast it?